test/tuntap-unit.spec.js

```diff
@@ -0,0 +1,78 @@
+import assert from "assert";
+import { TunTap } from "../lib/index.js";
+
+/**
+ * NOTE: Most TunTap tests require root privileges (sudo) to run.
+ * If not running as root, privileged tests will be skipped.
+ */
+
+describe("TunTap Unit Tests", function () {
+  let tun;
+  let isRoot = typeof process.getuid === "function" ? process.getuid() === 0 : false;
```

Comment:
it probably makes sense to extract this method into common test utils, so it could be reused

Reply:
updted with commit https://github.com/appium/appium-ios-tuntap/pull/1/commits/bcb9e8623f1bc9c54f8dfeb8e9c645d58dcfad4d